### PR TITLE
fix(coverage): use source maps when printing pretty reports

### LINKED ISSUE #8923: Report source mapped code coverages
For the initial pass I did straight JavaScript code coverage as the runtime saw it for simplicity of implementation. This was never intended to be permanent and we should match the user's expectation and render the mapped source in reports.

### PR DESCRIPTION
**Pull request by caspervonb**
This makes use of source maps and the original source when printing lacking line coverage in the pretty printer.

Only the executable lines are checked as before (as non-executable lines will always be ignored anyways).
The lines then mapped to the appropriate source line when a source map is present.

Closes #8923

### CODE REVIEW

File: cli/tools/coverage.rs [hunk 1]
```diff
@@ -221,8 +231,34 @@ impl PrettyCoverageReporter {
...
+        } else {
+          lines
+        };
+
+      let output_indices = if let Some(source_map) = maybe_source_map {
+        let mut indices = uncovered_lines
+          .iter()
+          .map(|i| {
+            source_map
+              .lookup_token(*i as u32, 16)
```
Comment (caspervonb) [inline cli/tools/coverage.rs:246]: Magical offset; lookup_token searches backwards so it'll find something NEAR the line; not the actual line.

File: cli/tools/coverage.rs [hunk 2]
```diff
@@ -221,8 +231,34 @@ impl PrettyCoverageReporter {
         println!("{}", colors::red(&line_coverage));
       }
 
+      let output_lines =
+        if let Some(original_source) = maybe_original_source.as_ref() {
+          original_source.split('\n').collect::<Vec<_>>()
```
Comment (ry) [inline cli/tools/coverage.rs:236]: Does this handle CRLF files correctly?

File: cli/tests/test_run_test_coverage.out [hunk 3]
```diff
@@ -10,23 +10,24 @@ ok ([WILDCARD])
...
-cover [WILDCARD]/subdir/subdir2/mod2.ts ... 62.500% (5/8)
-   5 | export function printHello2() {
-   6 |     printHello();
-   7 | }
-cover [WILDCARD]/test_coverage.ts ... 100.000% (5/5)
+cover [WILDCARD]/tests/subdir/subdir2/mod2.ts ... 62.500% (5/8)
+   7 | export function printHello2(): void {
+   8 |   printHello();
+   9 | }
+cover [WILDCARD]/tests/test_coverage.ts ... 100.000% (5/5)
```
Comment (ry) [inline cli/tests/test_run_test_coverage.out:33]: Nice!

File: cli/tools/coverage.rs (hunk 2 above)
Comment (caspervonb) [reply cli/tools/coverage.rs:236]: Right, windows.

I believe so, we also do the same in program_state (https://github.com/denoland/deno/blob/master/cli/program_state.rs#L324).

File: cli/tools/coverage.rs (hunk 1 above)
Comment (caspervonb) [inline cli/tools/coverage.rs:246]: 1:1 mapping also isn't going to work; should get all the source lines that the executed line represents (happens to be 1:1 in most of our generated code but brittle).

**Review by bartlomieju — APPROVED** [review]
LGTM, great to have this fixed!

**Pull request merged**